types/config.go

```diff
@@ -0,0 +1,108 @@
+package types
+
+import (
+	"sync"
+)
+
+// Config is the structure that holds the SDK configuration parameters
+// This could be used to initialize certain configuration parameters for the SDK
+type Config struct {
```

Comment:
This is great! My only fear is this may become a dumping ground for many things. Perhaps this is fine for now but we should be cautious.

Reply:
It is a scary pattern. I'm not sure if there is a better one.

Maybe it could be added to the ctx object?

Reply:
I'm not sure what is the best way to proceed here than having a global config. Passing context seems more reasonable but I think functions in types/address.go needs to allow getting context and this needs more refactoring. Or, should we hold this change until types/address.go is refactored?

Reply:
I don't think this should go in the context. Its really unique per app, not unique per state transition / block. For now, I think we should merge this. The real problem imo is this monolithic "types" package (which he have several issues collobarating)

I think this is fine for now :)

Reply:
Thanks @ValarDragon for the clarification. Just pushed updated changes.

Reply:
yeah this is good for now. Definitely no to a context.